### PR TITLE
Feature/add data map

### PR DESCRIPTION
**Pull request by suho**
- Finished Basic App

### CODE REVIEW

File: FourSquare/Controllers/Base/PageBase/MenuItemViewController.swift [hunk 1]
```diff
@@ -60,11 +70,62 @@ class MenuItemViewController: BaseViewController {
...
+    private func setUpRefreshControl() {
+        self.refreshControl = UIRefreshControl()
+        self.refreshControl.addTarget(self, action: #selector(self.refreshData), forControlEvents: .ValueChanged)
+        self.venueTableView?.addSubview(refreshControl)
+    }
+
+    func refreshData() {
+        self.offset = 0
+        self.deleteVenues()
+        if self.section == .Search {
```
Comment (le-binh) [inline FourSquare/Controllers/Base/PageBase/MenuItemViewController.swift:82]: should override this function

File: FourSquare/Controllers/Base/PageBase/MenuItemViewController.swift [hunk 2]
```diff
@@ -60,11 +70,62 @@ class MenuItemViewController: BaseViewController {
...
+    }
+
+    private func loadVenues() {
+        SVProgressHUD.show()
+        self.refreshControl.endRefreshing()
+        VenueService().loadVenues(section.rawValue, limit: self.limit, offset: self.offset) { (venues) in
+            SVProgressHUD.dismiss()
+            self.venueTableView?.reloadData()
+            if let delegate = self.delegate {
+                delegate.menuItemDidLoadData(self.venues)
```
Comment (le-binh) [inline FourSquare/Controllers/Base/PageBase/MenuItemViewController.swift:106]:
```
delegate?.menuItemDidLoadData(self.venues)
```

File: FourSquare/Controllers/Base/PageBase/MenuItemViewController.swift [hunk 3]
```diff
@@ -101,12 +162,15 @@ extension MenuItemViewController: UITableViewDelegate {
...
 
-    func scrollViewWillBeginDragging(scrollView: UIScrollView) {
-
-    }
-
     func scrollViewDidEndDecelerating(scrollView: UIScrollView) {
-
+        let currentOffset = scrollView.contentOffset.y
+        let maximumOffset = scrollView.contentSize.height - scrollView.frame.height
+        if maximumOffset - currentOffset < 2 * self.rowHeight && willLoadMore {
```
Comment (le-binh) [inline FourSquare/Controllers/Base/PageBase/MenuItemViewController.swift:168]: check for `willLoadMore` first for increasing performance

File: FourSquare/Controllers/Base/PageBase/MenuItemViewController.swift (hunk 1 above)
Comment (suho) [inline FourSquare/Controllers/Base/PageBase/MenuItemViewController.swift:82]: fixed

File: FourSquare/Controllers/Base/PageBase/MenuItemViewController.swift (hunk 2 above)
Comment (suho) [inline FourSquare/Controllers/Base/PageBase/MenuItemViewController.swift:106]: fixed

**Pull request merged**